File: ui/article-aem-renderer/src/main/java/org/cru/godtools/article/aem/api/AemApi.java
package org.cru.godtools.article.aem.api;

import android.net.Uri;

import org.json.JSONObject;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.Streaming;
import retrofit2.http.Url;

public interface AemApi {
    @GET
    Call<JSONObject> getJson(@Url Uri uri, @Query("_") long timestamp);

    @GET
    Call<String> downloadArticle(@Url Uri uri);

    @GET
    @Streaming
    Call<ResponseBody> downloadResource(@Url Uri uri);
}

File: ui/article-aem-renderer/src/main/java/org/cru/godtools/article/aem/api/AemApi.kt
package org.cru.godtools.article.aem.api

import android.net.Uri
import okhttp3.ResponseBody
import org.json.JSONObject
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Streaming
import retrofit2.http.Url

interface AemApi {
    @GET
    suspend fun getJson(@Url uri: Uri, @Query("_") timestamp: Long): Response<JSONObject?>

    @GET
    suspend fun downloadArticle(@Url uri: Uri): Response<String?>

    @GET
    @Streaming
    suspend fun downloadResource(@Url uri: Uri): Response<ResponseBody?>
}
